packages/bulkapi/src/controllers/BulkApi.ts

```diff
@@ -31,76 +32,105 @@ export class BulkApiImpl implements BulkApi {
                     }
                 }
-                if (node.annotations.length !== 0) {
-                    issues.push({ kind: "PartitionHasAnnotations", message: `Node ${node.id} cannot be created as a partition because it has annotations`} )
+                if (issues.length !== 0) {
+                    lionwebResponse<CreatePartitionsResponse>(response, HttpClientErrors.PreconditionFailed, {
+                        success: false,
+                        messages: issues
+                    })
+                    return
                 }
+                if (chunk.nodes.length === 0) {
+                    // do nothing, no new partitions
+                    lionwebResponse<CreatePartitionsResponse>(response, HttpSuccessCodes.Ok, {
+                        success: true,
+                        messages: [{ kind: "EmptyChunk", message: "-- empty partitions list, no query" }]
+                    })
+                    return
+                }
+                const x = await this.ctx.bulkApiWorker.createPartitions(clientId, chunk)
+                lionwebResponse<CreatePartitionsResponse>(response, x.status, x.queryResult)
             }
-            if (issues.length !== 0) {
-                lionwebResponse<CreatePartitionsResponse>(response, HttpClientErrors.PreconditionFailed, {
-                    success: false,
-                    messages: issues
-                })
-                return
-            }
-            if (chunk.nodes.length === 0) {
-                // do nothing, no new partitions
-                lionwebResponse<CreatePartitionsResponse>(response, HttpSuccessCodes.Ok, {
-                    success: true,
-                    messages: [{ kind: "EmptyChunk", message: "-- empty partitions list, no query"}]
-                })
-                return
-            }
-            const x = await this.ctx.bulkApiWorker.createPartitions(chunk)
-            lionwebResponse<CreatePartitionsResponse>(response, x.status, x.queryResult)
         }
     }
 
-    deletePartitions = async (req: Request, response: Response): Promise<void> => {
-        logger.requestLog(` * deletePartitions request received, with body of ${req.headers["content-length"]} bytes`)
-        const idList = req.body
-        const x = await this.ctx.bulkApiWorker.deletePartitions(idList)
-        lionwebResponse<DeletePartitionsResponse>(response, x.status, x.queryResult)
+    deletePartitions = async (request: Request, response: Response): Promise<void> => {
+        logger.requestLog(` * deletePartitions request received, with body of ${request.headers["content-length"]} bytes`)
+        const clientId = getStringParam(request, "clientId")
+        const idList = request.body
+        if (isResponseMessage(clientId)) {
+            logger.requestLog("STORE CLIENT ID ERROR: clientId incorrect: " + JSON.stringify(clientId))
```

Comment:
Would it make sense to automatically log errors every time we return a response with success = false?